### PR TITLE
GEOS-8947 gwc cluster safe via events

### PR DESCRIPTION
**Pull request by NielsCharlier**

### CODE REVIEW

File: src/gwc/src/main/java/org/geoserver/gwc/layer/DefaultTileLayerCatalog.java [hunk 1]
```diff
@@ -227,9 +254,72 @@ public GeoServerTileLayerInfo save(final GeoServerTileLayerInfo newValue) {
...
+                                listener.onEvent(tileLayerId, TileLayerCatalogListener.Type.DELETE);
+                            }
+                        }
+                    }
+                };
+        listenersByFileName.put(file.name(), existingLayerListener);
+        file.addListener(existingLayerListener);
+    }
+
+    private void stopToListen(Resource file) {
```
Comment (smithkm) [inline src/gwc/src/main/java/org/geoserver/gwc/layer/DefaultTileLayerCatalog.java]: This is a confusing name.  It suggests some sort of synchronous pause/wait: "stop" in order to "listen". `stopListening` would be clearer.  Renaming the other method `startListening` would keep the pairing between them clear.

**Comment by NielsCharlier** [comment]
thanks @smithkm 
I have made the requested changes
The build still shows failed, but it is unrelated

**Comment by smithkm** [comment]
The failure is in the new test you added so it seems related to me
```
	at org.geoserver.gwc.layer.DefaultTileLayerCatalogTest.waitForFlag(DefaultTileLayerCatalogTest.java:239)
	at org.geoserver.gwc.layer.DefaultTileLayerCatalogTest.testEvents(DefaultTileLayerCatalogTest.java:218)
```

**Comment by aaime** [comment]
Picked from the current Travis build log:

```
Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.435 sec <<< FAILURE!
testEvents(org.geoserver.gwc.layer.DefaultTileLayerCatalogTest)  Time elapsed: 326 sec  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at org.geoserver.gwc.layer.DefaultTileLayerCatalogTest.waitForFlag(DefaultTileLayerCatalogTest.java:239)
	at org.geoserver.gwc.layer.DefaultTileLayerCatalogTest.testEvents(DefaultTileLayerCatalogTest.java:218)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
```

**Comment by NielsCharlier** [comment]
I'm so confused. This is what I see in the current travis build log:

Running org.geoserver.gwc.layer.DefaultTileLayerCatalogTest
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.083 sec

I only see a failure for wps-remote.
How is this possible?

**Comment by NielsCharlier** [comment]
@smithkm the test has now been properly fixed

**Pull request merged**